packages/js/components/src/experimental-select-tree-control/select-tree.tsx

```diff
@@ -48,6 +49,68 @@ export const SelectTree = function SelectTree( {
 	const [ isFocused, setIsFocused ] = useState( false );
 	const [ isOpen, setIsOpen ] = useState( false );
 
+	const recalculateInputValue = () => {
+		if ( onInputChange ) {
+			if ( ! props.multiple && props.selected ) {
+				onInputChange( ( props.selected as Item ).label );
+			} else {
+				onInputChange( '' );
+			}
+		}
+	};
+
+	const focusOnInput = () => {
+		(
+			document.querySelector( `#${ props.id }-input` ) as HTMLInputElement
+		 )?.focus();
+	};
+
+	const inputProps: any = {
+		className: 'woocommerce-experimental-select-control__input',
+		id: `${ props.id }-input`,
+		'aria-autocomplete': 'list',
+		'aria-controls': `${ props.id }-menu`,
+		autoComplete: 'off',
+		onFocus: () => {
+			if ( ! isOpen ) {
+				setIsOpen( true );
+			}
+			setIsFocused( true );
+		},
+		onBlur: ( event: any ) => {
+			// if blurring to an element inside the dropdown, don't close it
+			if (
+				isOpen &&
+				! document
+					.querySelector( '.' + menuInstanceId )
+					?.contains( event.relatedTarget )
```

Comment:
You could extract this to a function and do something like `! isInsideDropdown( event.relatedTarget )`... if you did that, you don't even need the comment... the code self-documents.